what4/src/What4/Expr/Builder.hs

```diff
@@ -1088,11 +1089,57 @@ sbConcreteLookup sym arr0 mcidx idx
     -- Lookups on constant arrays just return value
   | Just (ConstantArray _ _ v) <- asApp arr0 = do
       return v
-    -- Lookups on mux arrays just distribute over mux.
-  | Just (BaseIte _ _ p x y) <- asApp arr0 = do
-      xv <- sbConcreteLookup sym x mcidx idx
-      yv <- sbConcreteLookup sym y mcidx idx
-      baseTypeIte sym p xv yv
+
+    -- A lookup in an array copy is a lookup in the src array when inside the copy range
+  | Just (CopyArray w _a_repr _dest_arr dest_begin_idx src_arr src_begin_idx _len dest_end_idx _src_end_idx) <- asApp arr0
+  , Just (Empty :> (BVIndexLit _ lookup_idx_bv)) <- mcidx
+  , lookup_idx_unsigned <- BV.asUnsigned lookup_idx_bv
+  , Just dest_begin_idx_unsigned <- BV.asUnsigned <$> asBV dest_begin_idx
```

Comment:
I don't know if it is worth trying to do this, but you can sometimes get useful answers to range questions using the abstract domain computations even when the inputs you start with aren't fully concrete.